lib/cpp/src/thrift/transport/THeaderTransport.h

```diff
@@ -135,8 +141,9 @@ class THeaderTransport : public TVirtualTransport<THeaderTransport, TFramedTrans
   void transform(uint8_t* ptr, uint32_t sz);
 
   uint16_t getNumTransforms() const {
-    int trans = writeTrans_.size();
-    return trans;
+    std::vector<uint16_t>::size_type trans = writeTrans_.size();
+    assert(trans <= UINT16_MAX);
```

Comment:
I followed the pattern that already exists in other implementation files like THeaderTransport.cpp.  I agree with you here, I think it would be better to have code like:

```
if (trans > std::numeric_limits<uint16_t>().max()) { 
  throw TTransportException(CORRUPTED_DATA, ...); 
}
```

I think it would be better to open a new story to do this conversion across the codebase.  There have to be at least 10 of these checks in THeaderTransport.cpp alone.  Can we leave these as-is and address them holistically?

I did change the ones in the compiler code to use the existing pattern of throwing a std::string with an error message when something goes wrong.